pkg/kiln/docker_registry.go

```diff
@@ -0,0 +1,145 @@
+package kiln
+
+import (
+	"fmt"
+
+	"github.com/30x/kiln/pkg/registry"
+	"github.com/docker/engine-api/types"
+)
+
+//RegistryImageCreator is a struct that holds our pointer to the docker client
+type RegistryImageCreator struct {
+	//the client to docker
+	client registry.Registry
+	//the remote repository url
+	dockerCreator ImageCreator
+}
+
+//NewRegistryImageCreator creates an instance of the RegistryImageCreator from the docker environment variables, and returns the instance
+func NewRegistryImageCreator(repo string, reg registry.Registry) (ImageCreator, error) {
+	// necessary for using a GCR registry, the project name needs to be in the image tag
+	// and tagging is done by the LocalImageCreator before push
+	if proj := reg.GetProjectName(); proj != "" {
```

Comment:
This is reason one for separating use of a registry into Private Docker (docker_private) and GCR (docker_gcr)...thoughts?